### PR TITLE
ADD: exception controller

### LINKED ISSUE #236: Custom 404 page

### LINKED ISSUE #254: Deal with errors that arise from Tomcat upload limit
The default tomcat upload limit is 10MB, but the error is internal and if you try to upload a bigger file it just loads and loads and loads... until the browser's timeout is hit - while the server-side console flows over with exceptions.

TODO: show a useful error

### PR DESCRIPTION
**Pull request by realemx**
Closes #236 
Closes #254

### CODE REVIEW

File: src/main/java/mops/presentation/ExceptionController.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        ModelAndView mav = new ModelAndView("mops_error");
+        mav.getModel().put("error", ex);
+        mav.getModel().put("referrer", referrer);
+        return mav;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @RequestMapping("/error")
```
Comment (Segelzwerg) [inline src/main/java/mops/presentation/ExceptionController.java]: Is that restricted to our scope?

File: src/main/java/mops/presentation/ExceptionController.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            Integer statusCode = Integer.valueOf(status.toString());
+
+            if (statusCode == HttpStatus.NOT_FOUND.value()) {
+                error = "error-404";
+            } else if (statusCode == HttpStatus.INTERNAL_SERVER_ERROR.value()) {
+                error = "error-500";
+            }
+        }
+
+        return error;
```
Comment (Segelzwerg) [inline src/main/java/mops/presentation/ExceptionController.java]:
```suggestion
        if (status == null) {
            return error;
        }
        Integer statusCode = Integer.valueOf(status.toString());

        if (statusCode == HttpStatus.NOT_FOUND.value()) {
            error = "error-404";
        } else if (statusCode == HttpStatus.INTERNAL_SERVER_ERROR.value()) {
            error = "error-500";
         }
        
        return error;
```
maybe cleaner?

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -27,3 +27,9 @@ material1.mops.gruppenfindung.url=${MATERIAL1_GRUPPENFINDUNG_URL:https://mops.hh
 
 material1.mops.configuration.admin=${MATERIAL1_ADMIN_ROLE:admin}
 material1.mops.configuration.max-groups=${MATERIAL_MAX_FOLDER_PER_GROUP:200}
+
+spring.servlet.multipart.enabled=true
+spring.servlet.multipart.file-size-threshold=2KB
+spring.servlet.multipart.max-file-size=${MATERIAL1_MAX_FILE_SIZE:10MB}
+spring.servlet.multipart.max-request-size=${MATERIAL1_MAX_FILE_SIZE:10MB}
+server.tomcat.max-swallow-size=-1
```
Comment (Segelzwerg) [inline src/main/resources/application.properties]: Can you explain that in documentation?

File: src/main/resources/templates/error-404.html [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html lang="en" xmlns:th="http://www.thymeleaf.org"
+      th:replace="~{mopslayout :: html(name='Name des Subsystems', title='Optional: Tabtitel', headcontent=~{:: headcontent}, navigation=~{:: navigation}, bodycontent=~{:: bodycontent})}">
```
Comment (Segelzwerg) [inline src/main/resources/templates/error-404.html]: Can you fill that.

File: src/main/java/mops/presentation/ExceptionController.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    @SuppressWarnings("PMD.LawOfDemeter")
+    public ModelAndView resolveException(HttpServletRequest request,
+                                         HttpServletResponse response,
+                                         Object handler, Exception ex) {
+
+        String referrer = request.getHeader("referer");
+        ModelAndView mav = new ModelAndView("mops_error");
+        mav.getModel().put("error", ex);
+        mav.getModel().put("referrer", referrer);
+        return mav;
```
Comment (iTitus) [inline src/main/java/mops/presentation/ExceptionController.java]: can you just return "mops_error" and use a Model to put the attributes there

File: src/main/java/mops/presentation/ExceptionController.java (hunk 2 above)
Comment (iTitus) [reply src/main/java/mops/presentation/ExceptionController.java]: do we need three different templates that are basically the same except for a little text?

File: src/main/resources/application.properties (hunk 3 above)
Comment (iTitus) [reply src/main/resources/application.properties]: needs new line

File: src/main/resources/templates/mops_error.html [hunk 6]
```diff
@@ -18,6 +18,7 @@
     <div class="alert alert-warning" id="error-message" th:if="${error} != null" th:text="${error.message}">
         Error!
     </div>
+    <a th:if="${referrer}" th:href="${referrer}" class="nav-link">Zurück</a>
 </main>
 </body>
 </html>
```
Comment (iTitus) [inline src/main/resources/templates/mops_error.html]: new line
(my bad)

File: src/main/java/mops/presentation/ExceptionController.java (hunk 5 above)
Comment (realemx) [reply src/main/java/mops/presentation/ExceptionController.java]: nope

Comment (iTitus) [reply src/main/java/mops/presentation/ExceptionController.java]: this is an interface and this requires us to use ModelAndView

**Pull request merged**